### PR TITLE
perf(cdc): Remove dropped message metric from filter

### PR DESCRIPTION
**Pull request by tkaemming**
Consumers that use this filter spend a substantial amount of time waiting on `socket.send` for these individual metrics.

The filtration ratio for these consumers is a useful metric, but we will need to find a different method of reporting these metrics moving forward that aggregates the data in-process rather than sending each metric individually. (My hope is to add this in the future to the streaming package so that this is automatically instrumented on the step.)

In the meantime, this will allow us to get a better sense of the consumer performance without the overhead from recording this high density metric.

### CODE REVIEW

File: snuba/datasets/cdc/message_filters.py [hunk 1]
```diff
@@ -31,9 +31,7 @@ def should_drop(self, message: Message[KafkaPayload]) -> bool:
         )
 
         if table_name:
-            table_name = table_name.decode("utf-8")
-            if table_name != self.__postgres_table:
```
Comment (tkaemming) [inline snuba/datasets/cdc/message_filters.py:35]: (This was causing a str/bytes mismatch on the `table_name` local variable, so I collapsed it into the conditional.)

File: snuba/datasets/cdc/message_filters.py [hunk 2]
```diff
@@ -31,9 +31,7 @@ def should_drop(self, message: Message[KafkaPayload]) -> bool:
         )
 
         if table_name:
```
Comment (tkaemming) [inline snuba/datasets/cdc/message_filters.py:33]: If there is no table name, this method falls through to `False`, as in "this message should not be dropped."

This means that CDC events that are not associate with a table (i.e. begin, commit) are sent to the worker pool and require serialization and deserialization on both ends of that parallel transformation.

I will filter these out in a separate change, and I would expect this filter the stream by up to 2/3rds given our substantial usage of autocommit.

File: snuba/datasets/cdc/message_filters.py [hunk 3]
```diff
@@ -31,9 +31,7 @@ def should_drop(self, message: Message[KafkaPayload]) -> bool:
...
-            table_name = table_name.decode("utf-8")
-            if table_name != self.__postgres_table:
-                metrics.increment("cdc_message_dropped", tags={"table": table_name})
+            if table_name.decode("utf-8") != self.__postgres_table:
```
Comment (fpacifici) [inline snuba/datasets/cdc/message_filters.py:34]: Why not
`return table_name and table_name.decode("utf-8") != self.__postgres_table`

now that there is no other statement in this method?

**Review by fpacifici — APPROVED** [review]

File: snuba/datasets/cdc/message_filters.py (hunk 3 above)
Comment (tkaemming) [reply snuba/datasets/cdc/message_filters.py:34]: Good point, will tidy this up.

**Pull request merged**